src/odemis/driver/xt_client.py

```diff
@@ -1041,10 +1041,12 @@ def _setVoltage(self, voltage):
 
     def _setBlanker(self, blank):
         """True if the the electron beam should blank, False if it should be unblanked."""
+        self.parent.set_channel_state(DETECTOR2CHANNELNAME["se-detector"], True)
```

Comment:
@pieleric The problem is actually not in setting the blanker, but in getting the blanker. To get the true state of the beam blank, the stream needs to run. I can add a `_getBlanker` that looks like this:

``` python
def _getBlanker(self, blank):
    self.parent.set_channel_state(DETECTOR2CHANNELNAME["se-detector"], True)
    blanked = self.parent.beam_is_blanked()
    self.parent.set_channel_state(DETECTOR2CHANNELNAME["se-detector"], False)
    return blanked
```

The only thing I'm a bit hesitant about is that if we add this to `_updateSettings` the channel will then be unpaused/paused every 5 seconds when the Delmic XTAdapter is running. I'm not sure if users will like how that looks?

Reply:
Definitely, we don't want to do that in the getter. 
The change to the setter are not useful then? If so, remove them, and let's give up on this function properly working.
I'd suggest to just adjust the test cases to not complain if the blanking VA is not updated.

Reply:
The changes I made to the setter were not useful, I've reverted them. I also updated the test, so now at least it passes.